qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/features/add_feature.py

```diff
@@ -0,0 +1,32 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+
+class addFeature(object):
```

Comment:
- The class name does not match our standards, it should be `AddFeature`
- 
Assuming this should always be overriden, and nobody should ever call an instance of `addFeature` directly:

- This should be an [abstract base class](https://docs.python.org/3/library/abc.html#abc.ABC), and the methods that return nothing should really raise a NotImplementedError
- Some doc strings would be really nice, what is `add_to_tab` meant to do, setup the view, call disable, something else?